### PR TITLE
Update the SSO button design and fix autofocus

### LINKED ISSUE #36883: proper branding for SSO provider
### What would you like Teleport to do?
In the `Sign in To Teleport` page, use proper icon and branding for SSO tiles.

Screenshot 2024-01-18 at 2 20 55 PM

- Why is Google tile colored "red" and orange for Okta? 
- Missing Okta Icon.

### What problem does this solve?
Proper branding increases trust in the login page. 

### If a workaround exists, please include it.
n.a

### LINKED ISSUE #44874: [Update the SSO buttons](https://www.figma.com/design/v6GunK50D2VC7w7I2FBDNf/Management?node-id=3707-40817&m=dev#858768926)

### PR DESCRIPTION
**Pull request by bl-nero**
- Use correct IDP icons
- Add Okta to the list of guessed providers
- Fix broken autofocus for SSO and passwordless methods
- Update the docs that contained previously incorrect information about the display attribute of the connector resource definition

Screenshot 2024-07-31 at 19 25 03

Figma: https://www.figma.com/design/v6GunK50D2VC7w7I2FBDNf/Management?node-id=4044-8185&m=dev

Deviations from the design:
- We decided to abandon the "sign in with" prefix, as it may conflict with existing deployments.
- We decided to revert back to the centered text.

Fixes #44874
Fixes #36883

### CODE REVIEW

**Comment by bl-nero** [comment]
Oops, I've just realized I broke the layout on Connect. Will fix it tomorrow.

File: web/packages/design/src/ResourceIcon/index.tsx [hunk 1]
```diff
@@ -45,7 +45,9 @@ export const ResourceIcon = ({ name, ...props }: ResourceIconProps) => {
   if (!icon) {
     return null;
   }
-  return <Image src={icon} {...props} />;
+  // Note: we add the class name for consistency with `Icon`, where it's used
+  // for testing.
+  return <Image src={icon} className={`res-icon-${name}`} {...props} />;
```
Comment (kimlisa) [inline web/packages/design/src/ResourceIcon/index.tsx:50]: i wrote that test when i was just starting out 😅, can we use data-testid instead?

File: web/packages/shared/components/ButtonSso/ButtonSso.test.tsx [hunk 2]
```diff
@@ -26,12 +26,13 @@ import ButtonSso from '.';
...
-  ${'github'}       | ${'icon-github'}
-  ${'bitbucket'}    | ${'icon-key'}
-  ${'google'}       | ${'icon-google'}
+  ${'Microsoft'}    | ${'res-icon-microsoft'}
+  ${'github'}       | ${'res-icon-github'}
+  ${'bitbucket'}    | ${'res-icon-atlassianbitbucket'}
+  ${'google'}       | ${'res-icon-google'}
 `('rendering of $ssoType', ({ ssoType, expectedIcon }) => {
   render(<ButtonSso ssoType={ssoType} title="hello" />);
+  screen.debug();
```
Comment (kimlisa) [inline web/packages/shared/components/ButtonSso/ButtonSso.test.tsx:35]: leftover?

**Review by kimlisa — APPROVED** [review]

File: web/packages/shared/components/ButtonSso/ButtonSso.test.tsx (hunk 2 above)
Comment (bl-nero) [reply web/packages/shared/components/ButtonSso/ButtonSso.test.tsx:35]: Yes, indeed.

**Comment by bl-nero** [comment]
Cluster connect dialog layout fixed.

Screenshot 2024-08-01 at 12 46 11

File: web/packages/teleport/src/components/FormLogin/SsoButtons.tsx [hunk 3]
```diff
@@ -21,25 +21,27 @@ import { Flex, Text } from 'design';
...
-          title={title}
-          ssoType={ssoType}
-          disabled={isDisabled}
-          autoFocus={index === 0 && autoFocus}
-          onClick={e => {
-            e.preventDefault();
-            onClick(item);
-          }}
-        />
+        <React.Fragment key={index}>
```
Comment (gzdunek) [inline web/packages/teleport/src/components/FormLogin/SsoButtons.tsx:31]: A leftover? It doesn't do anything, no?

**Review by gzdunek — APPROVED** [review]

**Pull request merged**